### DISCUSSION TITLE
Adding Reddit as a Login Provider?

### DISCUSSION THREAD

**Discussion by stnrd** [pocketbase/pocketbase#1851] [Show and tell]
Hi all, 

I added Reddit as a login provider to a fork of Pocketbase, but I discovered that Reddit is not a true single sign-on provider, as it is intended to be used for building on top of its platform. For instance, the sessions only last for one hour, and Reddit doesn't share user email addresses. Despite this, I was able to successfully add Reddit as a login provider and it is currently functioning properly. 

I'm curious to know if this integration is desired in Pocketbase, and if so, I can create a pull request with the changes. The branch with the changes can be found at https://github.com/stnrd/pocketbase/tree/feature/add-reddit-login.

Kind regards, 
Sten

**Comment by ganigeorgiev** [comment]
Contributions for new OAuth2 providers are welcome as long as the provider has an official OAuth2 API (_ideally with PKCE flow support, but that is not required and it will be less important in the near future with #55_).

> For instance, the sessions only last for one hour, and Reddit doesn't share user email addresses.

That's OK with PocketBase. The only requirement at the moment from the OAuth2 provider is to return a unique user identifier.
The provider session duration doesn't really matter because once the OAuth2 user is returned we create/link a PocketBase user (_in the worst case scenario, next time the client want to sing-in with Reddit they'll just have to go through the Reddit login screen_).
Email is also no longer required (_the constraint can be enforced from the auth collection options and if the OAuth2 provider doesn't return an email, it will just throw an error on sign-up_).

Feel free to open a PR with the suggested implementation and I'll review it.
